### ISSUE TITLE
Error in error showing

### ISSUE THREAD

**Issue by maleadt** [JuliaCI/Nanosoldier.jl#72]
```
┌ Error: [Node 2 | 2020-05-11T00:21:15.216]: Encountered error: encountered job loop error: MethodError(Base.show_backtrace, (IOBuffer(data=UInt8[...], readable=true, writable=true, seekable=true, append=false, size=1113, maxsize=Inf, ptr=1114, mark=-1), nothing), 0x00000000000068be)
│   exception =
│    MethodError: no method matching show_backtrace(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Nothing)
│    Closest candidates are:
│      show_backtrace(::IO, !Matched::Array{Any,1}) at errorshow.jl:595
│      show_backtrace(::IO, !Matched::Array{T,1} where T) at errorshow.jl:567
│    Stacktrace:
│     [1] #showerror#656(::Bool, ::typeof(showerror), ::Base.GenericIOBuffer{Array{UInt8,1}}, ::ProcessFailedException, ::Nothing) at ./errorshow.jl:79
│     [2] showerror(::Base.GenericIOBuffer{Array{UInt8,1}}, ::ProcessFailedException, ::Nothing) at ./errorshow.jl:74
│     [3] showerror(::Base.GenericIOBuffer{Array{UInt8,1}}, ::TaskFailedException) at ./task.jl:76
│     [4] showerror(::Base.GenericIOBuffer{Array{UInt8,1}}, ::CompositeException) at ./task.jl:49
│     [5] show at /home/maleadt/Nanosoldier/src/Nanosoldier.jl:44 [inlined]
│     [6] print(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Nanosoldier.NanosoldierError{CompositeException}) at ./strings/io.jl:35
│     [7] print_to_string(::Nanosoldier.NanosoldierError{CompositeException}) at ./strings/io.jl:138
│     [8] string(::Nanosoldier.NanosoldierError{CompositeException}) at ./strings/io.jl:177
│     [9] delegate_job(::Nanosoldier.Server, ::Nanosoldier.PkgEvalJob, ::Int64) at /home/maleadt/Nanosoldier/src/server.jl:108
│     [10] (::Nanosoldier.var"#55#56"{Nanosoldier.Server,DataType,Int64,Array{Int64,1}})() at /home/maleadt/Nanosoldier/src/server.jl:68
└ @ Nanosoldier ~/Nanosoldier/src/config.jl:51
```

**Comment by vtjnash** [comment]
I don't remember all the specifics, but we've likely fixed this in Julia v1.6
```
julia> throw(TaskFailedException(@sync @async ()))
ERROR: TaskFailedException
Stacktrace:
 [1] top-level scope
   @ task.jl:384

    nested task error: 

julia> throw(TaskFailedException(@sync @async error()))
ERROR: TaskFailedException

    nested task error: 
    Stacktrace:
     [1] error()
       @ Base ./error.jl:42
     [2] (::var"#7#8")()
       @ Main ./task.jl:406
Stacktrace:
 [1] sync_end(c::Channel{Any})
   @ Base ./task.jl:364
 [2] top-level scope
   @ task.jl:383
```
vs
```
julia> throw(TaskFailedException(@sync @async ()))
ERROR: TaskFailedException:
nothing
Stacktrace:
 [1] top-level scope at task.jl:334
SYSTEM (REPL): showing an error caused an error
ERROR: MethodError: no method matching show_backtrace(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Nothing)
Closest candidates are:
  show_backtrace(::IO, ::Array{Any,1}) at errorshow.jl:667
  show_backtrace(::IO, ::Array{T,1} where T) at errorshow.jl:639
Stacktrace:
 [1] showerror(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Nothing, ::Nothing; backtrace::Bool) at ./errorshow.jl:93
 [2] showerror(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Nothing, ::Nothing) at ./errorshow.jl:88
 [3] showerror(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::TaskFailedException) at ./task.jl:76
 [4] (::Base.var"#689#690"{TaskFailedException})(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}) at ./errorshow.jl:90
 [5] with_output_color(::Function, ::Symbol, ::IOContext{REPL.Terminals.TTYTerminal}; bold::Bool) at ./util.jl:72
 [6] with_output_color(::Function, ::Symbol, ::IOContext{REPL.Terminals.TTYTerminal}) at ./util.jl:70
 [7] showerror(::IOContext{REPL.Terminals.TTYTerminal}, ::TaskFailedException, ::Array{Base.StackTraces.StackFrame,1}; backtrace::Bool) at ./errorshow.jl:89
 [8] show_exception_stack(::IOContext{REPL.Terminals.TTYTerminal}, ::Array{Any,1}) at ./errorshow.jl:737
 [9] display_error(::IOContext{REPL.Terminals.TTYTerminal}, ::Array{Any,1}) at ./client.jl:104
 [10] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [11] invokelatest at ./essentials.jl:709 [inlined]
 [12] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:233
 [13] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:223
 [14] (::REPL.var"#do_respond#54"{Bool,Bool,REPL.var"#64#73"{REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:822
 [15] (::REPL.var"#69#78")(::Any, ::Any, ::Vararg{Any,N} where N) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1086
 [16] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [17] invokelatest at ./essentials.jl:709 [inlined]
 [18] (::REPL.LineEdit.var"#22#23"{REPL.var"#69#78",String})(::Any, ::Any) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:1364
 [19] prompt!(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2447
 [20] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2350
 [21] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /data/vtjnash/julia15/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1144
 [22] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:356

julia> throw(TaskFailedException(@sync @async error()))
ERROR: TaskFailedException:

Stacktrace:
 [1] error() at ./error.jl:42
 [2] (::var"#5#6")() at ./task.jl:356
Stacktrace:
 [1] sync_end(::Channel{Any}) at ./task.jl:314
 [2] top-level scope at task.jl:333
```

**Comment by maleadt** [comment]
Ah great. Guess I'll have to upgrade PkgEval then, it's currently still using Julia 1.3. There used to be issues with the Nanosoldier codebase running on 1.5, but IIRC @christopher-dG has already upgraded Julia on the Nanosoldier nodes?
